packages/testlab/src/validate-api-spec.ts

```diff
@@ -3,20 +3,15 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import * as SwaggerParser from 'swagger-parser';
-import {OpenApiSpec} from '@loopback/openapi-spec';
+import {OpenApiSpec} from '@loopback/openapi-spec-types';
+const validator = require('swagger2openapi/validate.js');
+import * as util from 'util';
+const promisify = util.promisify;
+const promisifiedValidator = promisify(validator.validate);
 
 export async function validateApiSpec(spec: OpenApiSpec): Promise<void> {
-  const opts: SwaggerParser.Options = {
-    $refs: {
-      internal: false,
-      external: false,
-    },
-  } as SwaggerParser.Options;
-
-  // workaround for unhelpful message returned by SwaggerParser
-  // TODO(bajtos) contribute these improvements to swagger-parser
-  if (!spec.swagger) {
+  const opts = {};
+  if (!spec.openapi) {
```

Comment:
Are these explicit checks still required? I added them only because swagger-parser was reporting unhelpful errors in these edge cases.

Reply:
just copy paste them from the old swagger check :-p I thought they are pre-check to prevent the complicated schema validation afterwards.

honestly...I haven't got time to check the edge cases missing in v3 validator. I will do a quick spike in the subsequent PR.